numpy/core/src/multiarray/methods.c

```diff
@@ -373,6 +374,18 @@ PyArray_GetField(PyArrayObject *self, PyArray_Descr *typed, int offset)
         }
         Py_DECREF(safe);
     }
+    self_elsize = PyArray_ITEMSIZE(self);
+    typed_elsize = typed->elsize;
+
+    /* check that values are valid */
+    if (typed_elsize > self_elsize) {
+        PyErr_SetString(PyExc_ValueError, "type too large");
+        return NULL;
+    }
+    if (offset < 0 || offset > self_elsize - typed_elsize) {
+        PyErr_SetString(PyExc_ValueError, "offset exceeds itemsize delta");
```

Comment:
`itemsize delta` won't mean anything to someone not looking at the code, it is more `offset too large`, although the first message is also a bit cryptic. So maybe `new type is larger than original type` and `new type plus offset is larger than original type`. Is there any guarantee that the types are aligned with each other?

Reply:
OK, rephrasing and I will also emit a different error for `offest < 0`

> Is there any guarantee that the types are aligned with each other?

Hmm. Do you mean there might be an internal offset in a dtype?